src/ui/reducers/sources.ts

```diff
@@ -328,6 +328,27 @@ export function getHasSiblingOfSameName(state: UIState, source: MiniSource) {
   return state.experimentalSources.sourcesByUrl[source.url]?.length > 0;
 }
 
+export const getCanonicalSourceFromEntities = (
+  detailsEntities: Dictionary<SourceDetails>,
+  sd: SourceDetails
+) => {
+  const canonicalSource = detailsEntities[sd.canonicalId];
+  return canonicalSource ?? sd;
```

Comment:
We might want to `console.error` or something if this isn't here. Unless I'm missing a case, that seems like it would mean something was really broken.